runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java

```diff
@@ -191,6 +198,49 @@ public void onChatMessage(ChatMessage event)
 			return;
 		}
 
+		if (event.getMessage().startsWith("The device is") && clue instanceof HotColdClue)
+		{
+			HotColdClue hotColdClue = (HotColdClue) clue;
+			String message = event.getMessage();
+			Matcher m1 = FINAL_STRANGE_DEVICE_MESSAGE.matcher(message);
+			Matcher m2 = STRANGE_DEVICE_MESSAGE.matcher(message);
+			Matcher m3 = INITIAL_STRANGE_DEVICE_MESSAGE.matcher(message);
+			// the order that these pattern matchers are checked is important
+			if (m1.find())
+			{
+				// final location for hot cold clue has been found
+				WorldPoint localWorld = client.getLocalPlayer().getWorldLocation();
+				if (localWorld != null)
+				{
+					hotColdClue.markFinalSpot(localWorld);
+				}
+			}
+			else if (m2.find())
+			{
+				String temperature = m2.group(1);
+				String difference = m2.group(2);
+				WorldPoint localWorld = client.getLocalPlayer().getWorldLocation();
+				if (localWorld != null)
+				{
+					hotColdClue.updatePossibleArea(localWorld, temperature, difference);
+				}
+			}
+			else if (m3.find())
+			{
+				String temperature = m3.group(1);
+				WorldPoint localWorld = client.getLocalPlayer().getWorldLocation();
+				if (localWorld != null)
+				{
+					hotColdClue.updatePossibleArea(localWorld, temperature, "");
+				}
+			}
+		}
+
+		if (event.getMessage().equals("The strange device cools as you find your treasure."))
```

Comment:
this check should be inverted and cause early return, otherwise continue to resetClue